modules/templates/helper.go

```diff
@@ -236,6 +236,12 @@ func NewFuncMap() []template.FuncMap {
 		"DisableWebhooks": func() bool {
 			return setting.DisableWebhooks
 		},
+		"DisableAccessTokens": func() bool {
+			return setting.DisableAccessTokens
+		},
+		"DisableOAuth2": func() bool {
```

Comment:
You don't mention anything about OAuth2 in this PR, what gives?

Reply:
> What is your potential reasoning and use-case for wanting to have these disabled?

If someone want to disable some auth options (like tokens) - should be allowed to do it (i.e. for security reasons). Other stuff like OAuth2 has switches.

> Realistically, almost no one will want to have it disabled as it breaks integration of almost any external service (including all CI/CD)

Think of systems that are isolated from all other apps and allows only authenticated user access (i.e. for security reasons).

All optional stuff in gitea should have switch to disable it. Don't try to be smarter than app owner.

Reply:
I find your security argument not convincing, considering that token authentication is far more secure than username/password combination for Basic Auth. Can you elaborate exactly how disabling this brings improved security over users being forced to specify (and remember) their username/password combinations in Git clients?

Reply:
> Can you elaborate exactly how disabling this brings improved security over users being forced to specify (and remember) their username/password combinations in Git clients?

Think of authentication with HTTP header using reverse proxy. In this scenario one may want to authenticate users ONLY with HTTP header from proxy and disable all other auth stuff (like ssh keys, passwords, basic auths and other shiny toys). Now its proxy job to make auth secure (SSO for example using secure auth methods).

Reply:
Git clients are already smart enough to use more advanced auth methods than username+password. Tokens should be left available as option of course it one needs it (i.e. external app that understands tokens only).

Reply:
> Think of authentication with HTTP header using reverse proxy. In this scenario one may want to authenticate users ONLY with HTTP header from proxy and disable all other auth stuff (like ssh keys, passwords, basic auths and other shiny toys). Now its proxy job to make auth secure (SSO for example using secure auth methods).

Assuming that Gitea can authenticate Git requests this way it would be a valid scenario.

Reply:
It can.